src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -210,6 +209,14 @@ public Response patchByDatasetUpdate(@Auth AuthUser authUser,
         throw new BadRequestException("Properties are invalid");
       }
       Dataset patched = datasetRegistrationService.patchDataset(datasetId, user, patch);
+      if (patched != null && patched.getIndexedDate() != null) {
```

Comment:
How can `patched` ever be `null`? If the DAO call fails, it will throw.

Reply:
The DAO call can return a null value from `return datasetDAO.findDatasetById(datasetId);` which does not throw a sql exception.